src/ipfs/schemas.ts

```diff
@@ -0,0 +1,101 @@
+
+export const coverValidatorsSchema = z.object({
+  version: z.literal(VERSION_1),
+  validators: z
+    .array(z.string().regex(ethereumAddressRegex, 'Invalid Ethereum address'))
+    .min(1, 'At least one validator address is required'),
+});
+
+export const coverQuotaShareSchema = z.object({
+  version: z.literal(VERSION_1),
+  quotaShare: z.number().min(0).max(100),
+});
+
+export const coverAumCoverAmountPercentageSchema = z.object({
+  version: z.literal(VERSION_1),
+  aumCoverAmountPercentage: z.number().min(0).max(100),
+});
+
+export const coverWalletAddressSchema = z.object({
+  version: z.literal(VERSION_1),
+  walletAddress: z.string().regex(ethereumAddressRegex, 'Invalid Ethereum address'),
+});
+
+export const coverWalletAddressesSchema = z.discriminatedUnion('version', [
+  z.object({
+    version: z.literal(VERSION_1),
+    walletAddresses: z.string().regex(ethereumAddressRegex, 'Invalid Ethereum address'),
+  }),
+  z.object({
+    version: z.literal(VERSION_2),
+    walletAddresses: z
+      .array(z.string().regex(ethereumAddressRegex, 'Invalid Ethereum address'))
+      .min(1, 'At least one wallet address is required'),
+  }),
+]);
+
+export const coverFreeTextSchema = z.object({
+  version: z.literal(VERSION_1),
+  freeText: z.string().min(1, 'Free text cannot be empty'),
+});
+
+export const coverDesignatedWalletsSchema = z.object({
+  version: z.literal(VERSION_1),
+  wallets: z
+    .array(
+      z.object({
+        wallet: z.string().regex(ethereumAddressRegex, 'Invalid Ethereum address'),
+        amount: z.string().regex(/^\d+$/, 'Amount must be a valid wei value'),
+      }),
+    )
+    .min(1, 'At least one wallet object is required'),
+});
+
+export const defiPassContentSchema = z.union([coverWalletAddressSchema, coverDesignatedWalletsSchema]);
+
+export const stakingPoolDetailsSchema = z.object({
+  version: z.literal(VERSION_1),
+  poolName: z.string().min(1, 'Pool name cannot be empty'),
+  poolDescription: z.string().min(1, 'Pool description cannot be empty'),
+});
+
+export const claimProofSchema = z.object({
+  version: z.literal(VERSION_1),
+  coverId: z.number().int().positive(),
+  affectedAddresses: z
+    .array(z.string().regex(ethereumAddressRegex, 'Invalid Ethereum address'))
+    .min(1, 'At least one affected address is required'),
+  affectedChain: z.string().min(1, 'Affected chain cannot be empty'),
+  incidentDescription: z.string().min(1, 'Incident description cannot be empty'),
+  incidentTransactionHashes: z
+    .array(z.string().min(1, 'Transaction hash cannot be an empty string'))
+    .min(1, 'At least one transaction hash is required'),
+  incidentEvidenceLinks: z
+    .array(z.string().url().min(1, 'Evidence link cannot be an empty string'))
+    .min(1, 'At least one evidence link is required'),
+  attachedFilesHashes: z
+    .array(z.string().min(1, 'Attached file hash cannot be an empty string'))
+    .min(1, 'At least one attached file hash is required'),
```

Comment:
would be good to have someone double check that these schemas align with what we want.

I've added `.min(1)` on most strings and arrays to ensure no empty array or strings are passed. is this what we want?

Reply:
Some are optional, depending on product type or other UI logic. This is going to be revisited anyway afaik - Lee mentioned something about it, and I'll discuss with him more when he gets back. Let's have them all optional (maybe keep the `incidentDescription` mandatory) for now, and we'll adjust them once we decide the final flow inputs.

Reply:
ok let's hold off this PR until we get confirmation from Lee exactly how these schemas should look like

Reply:
> but to unblock this, we can assume coverId and incidentDescription to be mandatory, the rest are optional
and affectedChain and affectedAddresses mandatory maybe

adding the conversation with @mixplore in Slack here for visibility